### PR TITLE
Allow creating proofs from compact props

### LINKED ISSUE #62: CreateProof should support literal and compact properties
The following method should be duplicated to allow both creation of proofs by literal names (what is implemented now) and compact property names. 

https://github.com/centrifuge/precise-proofs/blob/master/proofs/tree.go#L392

### PR DESCRIPTION
**Pull request by vimukthi-git**
fixes #62

### CODE REVIEW

**Comment by vimukthi-git** [comment]
@lucasvo I don't know why travis seems to run an older version of tests, do you know why?

File: proofs/tree.go [hunk 1]
```diff
@@ -389,15 +413,24 @@ func (doctree *DocumentTree) RootHash() []byte {
 }
 
 // CreateProof takes a property in dot notation and returns a Proof object for the given field
```
Comment (vimukthi-git) [inline proofs/tree.go]: need to adjust comment

**Comment by lucasvo** [comment]
Probably because there’s a conflict, try resolving it and test again?

**Comment by vimukthi-git** [comment]
@lucasvo thanks, it was indeed a conflict

File: proofs/tree.go [hunk 2]
```diff
@@ -211,6 +211,20 @@ type Salt struct {
 type Salts []Salt
 type GetSalt func(compact []byte) []byte
 
+type PropParam struct {
```
Comment (lucasvo) [inline proofs/tree.go]: is this really necessary? Why not just pass in a string or a byte slice when needed?

Comment (lucasvo) [reply proofs/tree.go]: It's only used in one method, CreateProof. Given an empty value is invalid, you could just check which of the two values are len()> 0 or we could create two `CreateProof` methods?

Comment (vimukthi-git) [reply proofs/tree.go]: The reason to create above type was to help handle any logic in conversions. But the conversions wasn't needed for now.
I can create two methods, 
- CreateProofForReadableProp
- CreateProofForCompactProp
And wrap `CreateProof` method using those, feels nicer as an interface. What do you think?

Comment (lucasvo) [reply proofs/tree.go]: I would prefer to remove the struct as it shouldn't be needed. And yes adding the two public methods make sense.

Comment (vimukthi-git) [reply proofs/tree.go]: done

File: proofs/tree.go [hunk 3]
```diff
@@ -420,26 +406,36 @@ func (doctree *DocumentTree) RootHash() []byte {
...
+
+	index, leaf := doctree.GetLeafByCompactProperty(prop)
+	if leaf == nil {
+		return proofspb.Proof{}, fmt.Errorf("No such field: %x in obj", prop)
+	}
+
+	return doctree.createProof(index, leaf)
+}
+
+// createProof takes a property in dot notation or compact form wrapped in a PropParam and returns a Proof object for the given field
```
Comment (lucasvo) [inline proofs/tree.go]: Incorrect comment

**Pull request merged**